prow/janitor.sh

```diff
@@ -9,164 +9,171 @@
 # Fail on any error.
 set -eo pipefail
 
-# Set the project id
-PROJECT="cloudesf-testing"
-REGION="us-central1"
-gcloud config set project ${PROJECT}
-gcloud config set run/region "${REGION}"
-
 # Resources older than 1 day should be cleaned up
 LIMIT_DATE=$(date -d "1 day ago" +%F)
 echo "Cleaning up resources before ${LIMIT_DATE}"
 
-### GKE Cluster ###
-GKE_SERVICES=$(gcloud container clusters list --format='value[separator=","](NAME,ZONE)' \
-  --filter="name ~ ^e2e- AND createTime  < ${LIMIT_DATE}" )
-
-for service in ${GKE_SERVICES};
-do
-  name=$(echo "${service}" | cut -d "," -f 1)
-  zone=$(echo "${service}" | cut -d "," -f 2)
-  echo "Deleting GKE service: name=${name}, zone=${zone}"
-  gcloud container clusters delete "${name}" \
-    --zone="${zone}" \
-    --async \
-    --quiet
-done
-echo "Done cleaning up GKE services"
-
-### Cloud Run ###
-CLOUD_RUN_SERVICES=$(gcloud run services list \
-    --platform=managed \
-    --filter="metadata.name ~ ^e2e-test- \
-    AND metadata.creationTimestamp < ${LIMIT_DATE}" \
-  --format="value(metadata.name)")
-
-for service in $CLOUD_RUN_SERVICES ; do
-  echo "Deleting Cloud Run service: ${service}"
-  gcloud run services delete "${service}" \
-    --platform=managed \
-    --quiet
-done
-echo "Done cleaning up Cloud Run services"
-
-### Cloud Functions ###
-GOOGLE_FUNCTIONS=$(gcloud functions list \
-    --filter="name ~ e2e-test- \
-    AND updateTime < ${LIMIT_DATE}" \
-  --format="value(name)")
-
-for gf in $GOOGLE_FUNCTIONS ; do
-  echo "Deleting Google Function: ${gf}"
-  gcloud functions delete ${gf} \
-    --quiet
-done
-echo "Done cleaning up Cloud Functions"
-
-### App Engines ###
-APP_ENGINES=$(gcloud app services list \
---filter="SERVICE ~ ^e2e-test-" \
---format="value(SERVICE)")
-for ap in ${APP_ENGINES} ; do
-  echo "Deleting App Engine: ${ap}"
-  gcloud app services delete "${ap}" \
-    --quiet
-done
-echo "Done cleaning up App Engines"
+# Set the project id
+PROJECT_IDS=("cloudesf-testing" "cloud-api-proxy-testing")
+REGIONS=("us-central1" "us-east1")
+for PROJECT in ${PROJECT_IDS[@]}; do
+  for REGION in ${REGIONS[@]}; do
+    echo "cleaning up resources in ${REGION} for ${PROJECT}"
```

Comment:
will we bail out the whole script if there is any CLI failed?

since cloud-esf-testing, only use us-cdentral1

cloud-api-proxy-testing only use us-east1

the other combinations may fail some CLI.

Reply:
Cool